src/KafkaFlow/Middlewares/Serializer/SerializerProducerMiddleware.cs

```diff
@@ -38,7 +38,7 @@ public async Task Invoke(IMessageContext context, MiddlewareDelegate next)
     {
         await _typeResolver.OnProduceAsync(context);
 
-        byte[] messageValue;
+        byte[] messageValue = null;
```

Comment:
Thank you for your PR.

This code will result in producing a byte[] value that is a string null when using the `JsonCoreSerializer`.

Therefore, as I said [here](https://github.com/Farfetch/kafkaflow/issues/516#issuecomment-1957371329), a message with a null value results in a tombstone record where the message value is an empty byte[].

Instead, you can consider the following:

```csharp
public async Task Invoke(IMessageContext context, MiddlewareDelegate next)
{
    await _typeResolver.OnProduceAsync(context);

    byte[] messageValue;

    if (context.Message.Value is not null)
    {
        using (var buffer = s_memoryStreamManager.GetStream())
        {
            await _serializer
                .SerializeAsync(
                    context.Message.Value,
                    buffer,
                    new SerializerContext(context.ProducerContext.Topic))
                .ConfigureAwait(false);

            messageValue = buffer.ToArray();
        }
    }

    await next(context.SetMessage(context.Message.Key, messageValue)).ConfigureAwait(false);
}
```

Just for you to know, if using a null, you might not be able to hit any TypeHandlers because Kafkaflow won't determine the valueType of the context message.

Reply:
I also think this could be a good addition to the documentation, can you add a paragraph to explain this here https://github.com/Farfetch/kafkaflow/blob/master/website/docs/guides/middlewares/serializer-middleware.md?